File: de.jwic.controls/src/de/jwic/controls/chart/impl/doughnut/DoughnutChart.js
#set($fld = $control.getField("content"))
{

	/**
	 * Invoked after the DOM element was updated. This function is NOT updated
	 * if the custom doUpdate function returned true.
	 */
	afterUpdate: function(element) {
		var field = jQuery(document.getElementById('${control.controlID}'));
		var ctx = document.getElementById('chart').getContext("2d");
		var randomScalingFactor = function(){ return Math.round(Math.random()*100)};
		Chart.defaults.global ={
				responsive : $control.global.responsive,
				animation: $control.global.animation,
				animationSteps: $control.global.animationSteps,
				animationEasing: '$control.global.animationEasing',
				showScale: $control.global.showScale,
				scaleOverride: $control.global.scaleOverride,
				scaleSteps: null,
				scaleStepWidth: null,
				scaleStartValue: null,
				 scaleLineColor: '$control.global.scaleLineColor',
				 scaleLineWidth: 1,
				 scaleShowLabels: true,
				 scaleLabel: "<%=value%>",
				 scaleIntegersOnly: true,
				 scaleBeginAtZero: false,
				 scaleFontFamily: "'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
				 scaleFontSize: 12,
				 scaleFontStyle: "normal",
				 scaleFontColor: "#666",
				 maintainAspectRatio: true,
				 showTooltips: true,
				 customTooltips: false,
				 tooltipEvents: ["mousemove", "touchstart", "touchmove"],
				 tooltipFillColor: "rgba(0,0,0,0.8)",
				 tooltipFontFamily: "'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
				 tooltipFontSize: 14,
				 tooltipFontStyle: "normal",
				 tooltipFontColor: "#fff",
				 tooltipTitleFontFamily: "'Helvetica Neue', 'Helvetica', 'Arial', sans-serif",
				 tooltipTitleFontSize: 14,
				 tooltipTitleFontStyle: "bold",
				  tooltipTitleFontColor: "#fff",
				  tooltipYPadding: 6,
				  tooltipXPadding: 6,
				  tooltipCaretSize: 8,
				  tooltipCornerRadius: 6,
				  tooltipXOffset: 10,
				  tooltipTemplate: "<%if (label){%><%=label%>: <%}%><%= value %>",
				  multiTooltipTemplate: "<%= value %>",
				  onAnimationProgress: function(){},
				  onAnimationComplete: function(){}


		}

			
			var options = {
				 // Boolean - Whether we should show a stroke on each segment
			    segmentShowStroke : true,

			    // String - The colour of each segment stroke
			    segmentStrokeColor : "#fff",

			    // Number - The width of each segment stroke
			    segmentStrokeWidth : 2,

			    // Number - The percentage of the chart that we cut out of the
				// middle
			    percentageInnerCutout : 50, // This is 0 for Pie charts

			    // Number - Amount of animation steps
			    animationSteps : 100,

			    // String - Animation easing effect
			    animationEasing : "easeOutBounce",

			    // Boolean - Whether we animate the rotation of the Doughnut
			    animateRotate : true,

			    // Boolean - Whether we animate scaling the Doughnut from the
				// centre
			    animateScale : false,

			    // String - A legend template
			    legendTemplate : "<ul class=\"<%=name.toLowerCase()%>-legend\"><% for (var i=0; i<segments.length; i++){%><li><span style=\"background-color:<%=segments[i].fillColor%>\"></span><%if(segments[i].label){%><%=segments[i].label%><%}%></li><%}%></ul>"

			};
			
		var data = [
		            {
		                value: 300,
		                color:"#F7464A",
		                highlight: "#FF5A5E",
		                label: "Red"
		            },
		            {
		                value: 50,
		                color: "#46BFBD",
		                highlight: "#5AD3D1",
		                label: "Green"
		            },
		            {
		                value: 100,
		                color: "#FDB45C",
		                highlight: "#FFC870",
		                label: "Yellow"
		            }
		        ]
		var myDoughnutChart = new Chart(ctx).Doughnut(data,options);


	}

	
	
}

File: de.jwic.controls/src/de/jwic/controls/chart/impl/doughnut/DoughnutChart.java
package de.jwic.controls.chart.impl.doughnut;

import java.awt.Color;

import de.jwic.base.IControlContainer;
import de.jwic.base.IncludeJsOption;
import de.jwic.base.JavaScriptSupport;
import de.jwic.controls.chart.api.Chart;
import de.jwic.controls.chart.api.ChartType;
import de.jwic.controls.chart.api.exception.ChartInconsistencyException;
import de.jwic.controls.chart.impl.util.DatenConverter;

/**
 * 
 * @author Karolina Marek (karolina-marek.eu)
 *
 * @date 29.11.2015
 */
@JavaScriptSupport
public class DoughnutChart extends Chart<DoughnutChartModel> {

	// Boolean - Whether we should show a stroke on each segment
	private boolean segmentShowStroke = true;

	// String - The colour of each segment stroke
	private Color segmentStrokeColor = new Color(0, 0, 0);
	// Number - The width of each segment stroke
	private int segmentStrokeWidth = 2;

	// Number - The percentage of the chart that we cut out of the
	// middle
	private int percentageInnerCutout = 50; // This is 0 for Pie charts

	// Number - Amount of animation steps
	private int animationSteps = 100;

	// String - Animation easing effect
	private String animationEasing = "easeOutBounce";

	// Boolean - Whether we animate the rotation of the Doughnut
	private boolean animateRotate = true;

	// Boolean - Whether we animate scaling the Doughnut from the
	// centre
	private boolean animateScale = false;

	// String - A legend template
	private String legendTemplate = "<ul class=\"<%=name.toLowerCase()%>-legend\"><% for (var i=0; i<segments.length; i++){%><li><span style=\"background-color=<%=segments[i].fillColor%>\"></span><%if(segments[i].label){%><%=segments[i].label%><%}%></li><%}%></ul>";

	public DoughnutChart(IControlContainer container, String name,
			DoughnutChartModel model) throws ChartInconsistencyException {
		super(container, name, ChartType.DOUGHNUT, model);

	}

	@IncludeJsOption(jsPropertyName = "segmentShowStroke")
	public boolean isSegmentShowStroke() {
		return segmentShowStroke;
	}

	public void setSegmentShowStroke(boolean segmentShowStroke) {
		this.segmentShowStroke = segmentShowStroke;
	}

	@IncludeJsOption(jsPropertyName = "segmentStrokeColor")
	public String getSegmentStrokeColor() {
		return DatenConverter.convertToJSColor(segmentStrokeColor);
	}

	public void setSegmentStrokeColor(Color segmentStrokeColor) {
		this.segmentStrokeColor = segmentStrokeColor;
	}

	@IncludeJsOption(jsPropertyName = "segmentStrokeWidth")
	public int getSegmentStrokeWidth() {
		return segmentStrokeWidth;
	}

	public void setSegmentStrokeWidth(int segmentStrokeWidth) {
		this.segmentStrokeWidth = segmentStrokeWidth;
	}

	@IncludeJsOption(jsPropertyName = "percentageInnerCutout")
	public int getPercentageInnerCutout() {
		return percentageInnerCutout;
	}

	public void setPercentageInnerCutout(int percentageInnerCutout) {
		this.percentageInnerCutout = percentageInnerCutout;
	}

	@IncludeJsOption(jsPropertyName = "animationSteps")
	public int getAnimationSteps() {
		return animationSteps;
	}

	public void setAnimationSteps(int animationSteps) {
		this.animationSteps = animationSteps;
	}

	@IncludeJsOption(jsPropertyName = "animationEasing")
	public String getAnimationEasing() {
		return animationEasing;
	}

	public void setAnimationEasing(String animationEasing) {
		this.animationEasing = animationEasing;
	}

	@IncludeJsOption(jsPropertyName = "scaleBeginAtZero")
	public boolean isAnimateRotate() {
		return animateRotate;
	}

	public void setAnimateRotate(boolean animateRotate) {
		this.animateRotate = animateRotate;
	}

	@IncludeJsOption(jsPropertyName = "animateScale")
	public boolean isAnimateScale() {
		return animateScale;
	}

	public void setAnimateScale(boolean animateScale) {
		this.animateScale = animateScale;
	}

	@Override
	public String getLegendTemplate() {

		return legendTemplate;
	}

	@Override
	public void setLegendTemplate(String legendTemplate) {
		this.legendTemplate = legendTemplate;
	}

}
